### ISSUE TITLE
✨ Add support for indenting wrapped lines

### ISSUE THREAD

**Issue by ben-p-commits** [CodeEditApp/CodeEditTextView#18]
## Is your feature request related to a problem? Please describe.
In the enhancement requested by https://github.com/CodeEditApp/CodeEdit/issues/868, the request to add configurable indentation to wrapped lines is beyond the current capabilities of `CodeEditTextView`.

<img width="368" alt="image" src="https://user-images.githubusercontent.com/5067237/210660792-41520a8e-73b7-49c2-9ee6-e238e91df619.png">

## Describe the solution you'd like
Add a parameter or option which allows custom indentation to be added for wrapped lines. May require enhancement  + PR in [STTextView](https://github.com/krzyzanowskim/STTextView)

## Additional Context

Example behavior in Xcode:

### 4 spaces
<img width="424" alt="image" src="https://user-images.githubusercontent.com/5067237/210662565-d071f269-9812-4722-a476-61588c07ff13.png">
<img width="527" alt="image" src="https://user-images.githubusercontent.com/5067237/210662737-9fd9644d-b77f-41ce-823d-1d6e2977a1bf.png">

### 12 spaces
<img width="430" alt="image" src="https://user-images.githubusercontent.com/5067237/210662296-fbb72de4-ba44-43b2-8471-38ffff73d79a.png">
<img width="432" alt="image" src="https://user-images.githubusercontent.com/5067237/210661748-ca871f41-35b0-4a4d-88c7-9e768f828523.png">

**Comment by austincondiff** [comment]
@Bogdan-Belogurov Where are we with this? No rush, just checking.

**Comment by Bogdan-Belogurov** [comment]
@austincondiff Hi! I'm still in progress. 
I'm trying to find the easiest way to implement this logic. I found this [PR](https://github.com/CodeEditApp/CodeEditTextView/pull/84) with `TextLayoutManagerRenderDelegate`, but the current `TextView` doesn't use this (only `MinimapView`). Am I right to add the realisation of the `TextLayoutManagerRenderDelegate` protocol to the `TextView`?

**Comment by thecoolwinter** [comment]
> @austincondiff @thecoolwinter Hi! I'm still in progress. 
> Am I right to add the realisation of the `TextLayoutManagerRenderDelegate` protocol to the `TextView`?

Not quite, that protocol is to allow developers to override layout or rendering behavior. CETV handles that by default but in the minimap's case it overrides how text is drawn (little dots).

Three places to look:
- `LineFragmentRenderer` does the actual drawing of the text in a graphics context. This will need to be modified to draw the indented fragments correctly.
- `Typesetter` determines line breaks - you'll want to modify this one to allow less width for line fragments that should be indented, which will change where line breaks occur.
- `LineFragment` will need to know about any indentation and adjust to account for that for instance when determining the content offset of an X coordinate in the fragment.

**Comment by alechash** [comment]
Hey @Bogdan-Belogurov and @thecoolwinter, not sure of the progress on this but if it has stalled I can take it. Could commit a few hours over the next two days + more this upcoming week.

Thanks, let me know!

**Comment by vksvicky** [comment]
Is anyone working on this? I can pick it up and have a look!

**Comment by alechash** [comment]
@vksvicky I would say go for it. I never heard back from them when I asked and by now I don’t have much free time at the moment to put anything toward it.

**Comment by vksvicky** [comment]
@alechash please assign this to me, so I can change the status and add relevant notes as I progress

**Comment by alechash** [comment]
@vksvicky I cannot assign this issue. Would need to come from @thecoolwinter.

**Comment by vksvicky** [comment]
Have made changes and here the pull request https://github.com/CodeEditApp/CodeEdit/pull/2158
